### ISSUE TITLE
The following threads are very CPU intensive

### ISSUE THREAD

**Issue by xinguangduan** [eclipse/jetty.project#6169]
**Jetty version**
9.4.31.v20200723
**Java version**
oracle jdk1.8.0_192
**OS type/version**
centos 
**Description**
The following threads are very CPU intensive

"qtp651235118-149433" #149433 prio=5 os_prio=0 tid=0x000000000261a800 nid=0x1e5c3d waiting on condition [0x00007fe03b6d5000]
java.lang.Thread.State: TIMED_WAITING (parking)
at sun.misc.Unsafe.park(Native Method)
- parking to wait for <0x000000056eb56c68> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2163)
at org.eclipse.jetty.util.SharedBlockingCallback$Blocker.block(SharedBlockingCallback.java:210)
at org.eclipse.jetty.server.HttpOutput.write(HttpOutput.java:217)
at org.eclipse.jetty.server.HttpOutput.close(HttpOutput.java:298)
at org.eclipse.jetty.server.Response.closeOutput(Response.java:1044)
at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:507)

qtp651235118-149434" #149434 prio=5 os_prio=0 tid=0x000000000749c000 nid=0x1e5c3e waiting on condition [0x00007fe050421000]
java.lang.Thread.State: TIMED_WAITING (parking)
at sun.misc.Unsafe.park(Native Method)
- parking to wait for <0x00000004fb25c910> (a java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject)
at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:2078)
at org.eclipse.jetty.util.BlockingArrayQueue.poll(BlockingArrayQueue.java:392)
at org.eclipse.jetty.util.thread.QueuedThreadPool.idleJobPoll(QueuedThreadPool.java:656)
at org.eclipse.jetty.util.thread.QueuedThreadPool.access$800(QueuedThreadPool.java:46)
at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:720)
at java.lang.Thread.run(Thread.java:748)

**Comment by joakime** [comment]
The 2 threads you pasted are parked / idle threads sitting in the ThreadPool waiting for some process to need them.
They use no CPU in that state.

They exist for as long as there is a need for them. (thread pool idle timeout will clean them out slowly based on non-use)
They can be used for any Jetty task that needs a thread and then put back into the ThreadPool.

**Comment by sbordet** [comment]
@xinguangduan the first thread is blocked in a write.

Like I said in the other comment, please upgrade to 9.4.39 -- we have done some work to perform closes in a non-blocking way.

**Comment by xinguangduan** [comment]
thanks for your reply,i will try the new version.

**Comment by xinguangduan** [comment]
![image](https://user-images.githubusercontent.com/17296491/114837911-c5c5d080-9e06-11eb-9d6e-315dee704414.png)
